### ISSUE TITLE
 osu!catch autoplay breaks on maps with a large amount of objects

### ISSUE THREAD

**Issue by juniebyte** [ppy/osu#3623]
The autoplay feature on osu!catch seems to break on maps with large amounts of objects (not confirmed, have only tested on one map)

I know this was meant to be a visualization map, but this bug could happen elsewhere, so I figured I should report it.

OS: Manjaro Linux (kernel 4.18.12-1-MANJARO)
osu!lazer version (manually built by me): 2018.1017.0
Beatmap: https://osu.ppy.sh/beatmapsets/150945

Steps to reproduce:

- open osu!lazer
- select osu!catch gamemode
- click play
- click solo
- select centipede beatmap
- go to mods and turn on autoplay
- play beatmap

It should crash right after the first "spinner" (don't know if they're called anything different in osu!catch, i typically only play the normal osu! gamemode)

Here was the error I got:

```
Unhandled Exception: System.ArgumentException: X must be finite, but is ∞.
   at osu.Framework.Graphics.Drawable.set_X(Single value)
   at osu.Game.Rulesets.Catch.UI.CatcherArea.UpdateAfterChildren() in /home/joyesh/osu/osu.Game.Rulesets.Catch/UI/CatcherArea.cs:line 104
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Game.Rulesets.UI.RulesetInputManager`1.UpdateSubTree() in /home/joyesh/osu/osu.Game/Rulesets/UI/RulesetInputManager.cs:line 141
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Platform.GameHost.UpdateFrame()
   at osu.Framework.Threading.GameThread.ProcessFrame()
   at osu.Framework.Threading.GameThread.runWork()
--- End of stack trace from previous location where exception was thrown ---
   at osu.Framework.Threading.Scheduler.Update()
   at osu.Framework.Threading.GameThread.ProcessFrame()
   at osu.Framework.Platform.GameHost.<>c__DisplayClass98_0.<Run>b__1(Object <p0>, FrameEventArgs <p1>)
   at OpenTK.GameWindow.RaiseUpdateFrame(Stopwatch watch, Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchUpdateFrame(Stopwatch watch)
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at osu.Framework.Platform.GameHost.Run(Game game)
   at osu.Desktop.Program.Main(String[] args) in /home/joyesh/osu/osu.Desktop/Program.cs:line 46
Aborted (core dumped)

**Comment by Hanamuke** [comment]
This is probably fixed by #3578

**Comment by peppy** [comment]
Please don't close issues until the related PR is also closed.

**Comment by peppy** [comment]
Please don't close issues until the related PR is also closed.

**Comment by ekrctb** [comment]
No longer reproducible. Probably fixed by #5035
